src/picker.rs

```diff
@@ -266,17 +264,129 @@ fn iterm2_from_env() -> Option<ProtocolType> {
-    rustix::termios::tcsetattr(stdin, OptionalActions::Drain, &termios)?;
+    termios::tcsetattr(&stdin, OptionalActions::Drain, &termios)?;
+
+    Ok(())
+}
+
+#[cfg(not(windows))]
+fn disable_raw_mode() -> Result<()> {
+    use rustix::termios::{self, OptionalActions};
+
+    if let Some(termios) = ORIGINAL_IN_MODE.lock().unwrap().as_ref() {
+        termios::tcsetattr(io::stdin(), OptionalActions::Now, termios)?;
+    }
+
+    Ok(())
+}
+
+#[cfg(not(windows))]
+fn font_size_fallback() -> Option<FontSize> {
+    use rustix::termios::{self, Winsize};
+
+    let winsize = termios::tcgetwinsize(io::stdout()).ok()?;
+    let Winsize {
+        ws_xpixel: x,
+        ws_ypixel: y,
+        ws_col: cols,
+        ws_row: rows,
+    } = winsize;
+    if x == 0 || y == 0 || cols == 0 || rows == 0 {
+        return None;
+    }
+
+    Some((x / cols, y / rows))
+}
+
+#[cfg(windows)]
+static ORIGINAL_IN_MODE: std::sync::Mutex<Option<windows::Win32::System::Console::CONSOLE_MODE>> =
+    std::sync::Mutex::new(None);
+
+#[cfg(windows)]
+use windows::Win32::Foundation::HANDLE;
+
+#[cfg(windows)]
+fn current_in_handle() -> Result<HANDLE> {
+    use windows::{
+        core::PCWSTR,
+        Win32::{
+            Foundation::{GENERIC_READ, GENERIC_WRITE, HANDLE},
+            Storage::FileSystem::{
+                self, FILE_FLAGS_AND_ATTRIBUTES, FILE_SHARE_READ, FILE_SHARE_WRITE, OPEN_EXISTING,
+            },
+        },
+    };
+
+    let utf16: Vec<u16> = "CONIN$\0".encode_utf16().collect();
+    let utf16_ptr: *const u16 = utf16.as_ptr();
+
+    Ok(unsafe {
+        FileSystem::CreateFileW(
+            PCWSTR(utf16_ptr),
+            (GENERIC_READ | GENERIC_WRITE).0,
+            FILE_SHARE_READ | FILE_SHARE_WRITE,
+            None,
+            OPEN_EXISTING,
+            FILE_FLAGS_AND_ATTRIBUTES(0),
+            HANDLE::default(),
+        )
+    }?)
+}
+
+#[cfg(windows)]
+fn enable_raw_mode() -> Result<()> {
+    use windows::Win32::System::Console::{
+        self, CONSOLE_MODE, ENABLE_ECHO_INPUT, ENABLE_LINE_INPUT, ENABLE_PROCESSED_INPUT,
+    };
+
+    let in_handle = current_in_handle()?;
+
+    let mut original_in_mode = CONSOLE_MODE::default();
+    unsafe { Console::GetConsoleMode(in_handle, &mut original_in_mode) }?;
+    *ORIGINAL_IN_MODE.lock().unwrap() = Some(original_in_mode);
+
+    let requested_in_modes = !ENABLE_ECHO_INPUT & !ENABLE_LINE_INPUT & !ENABLE_PROCESSED_INPUT;
+    let in_mode = original_in_mode & requested_in_modes;
+    unsafe { Console::SetConsoleMode(in_handle, in_mode) }?;
+
+    Ok(())
+}
+
+#[cfg(windows)]
+fn disable_raw_mode() -> Result<()> {
+    use windows::Win32::System::Console;
+
+    let in_handle = current_in_handle()?;
+
+    if let Some(in_mode) = ORIGINAL_IN_MODE.lock().unwrap().as_ref() {
+        unsafe { Console::SetConsoleMode(in_handle, *in_mode) }?;
+    }
+
+    Ok(())
+}
+
+#[cfg(windows)]
+fn font_size_fallback() -> Option<FontSize> {
+    Some((10, 20))
```

Comment:
Getting the font size through the API should be a better option. Unfortunately I didn't find a way to do this under Windows, but I think it's something that could be done. Of course in case the api fails, it would be nice to have a chance to specify the font size manually.